DotNetExcel/cardPurchase.cs

```diff
@@ -0,0 +1,219 @@
+
+            int lastRow = worksheet.Cells[worksheet.Rows.Count, "A"].End[XlDirection.xlUp].Row;
+            Range range = worksheet.Range["A" + TableFormat.HeaderRow + ":V" + lastRow];
+            range.ClearContents();
+
+            worksheet.Range["A" + TableFormat.HeaderRow].Value = "Data";
+            worksheet.Range["B" + TableFormat.HeaderRow].Value = "ID Compra";
+            worksheet.Range["C" + TableFormat.HeaderRow].Value = "Categoria";
+            worksheet.Range["D" + TableFormat.HeaderRow].Value = "Estabelecimento";
+            worksheet.Range["E" + TableFormat.HeaderRow].Value = "Descrição Compra";
+            worksheet.Range["F" + TableFormat.HeaderRow].Value = "Status";
+            worksheet.Range["G" + TableFormat.HeaderRow].Value = "Valor";
+            worksheet.Range["H" + TableFormat.HeaderRow].Value = "Anexo";
+            worksheet.Range["I" + TableFormat.HeaderRow].Value = "ID Cartão";
+            worksheet.Range["J" + TableFormat.HeaderRow].Value = "ID Holder";
+
+            Dictionary<string, object> optionalParam = new Dictionary<string, object>();
+
+            int row = TableFormat.HeaderRow + 1;
+
+            string cursor = "";
+            Dictionary<string, object> returnedData = new Dictionary<string, object>();
+
+            do
+            {
+
+                JObject respJson;
+
+                try
+                {
+                    respJson = corporatePurchase.Get(cursor);
+                }
+                catch (Exception ex)
+                {
+                    MessageBox.Show(ex.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+                    return;
+                }
+
+                if ((string)respJson["cursor"] != "") cursor = (string)respJson["cursor"];
+
+                JArray purchases = (JArray)respJson["purchases"];
+
+                foreach (JObject purchase in purchases)
+                {
+
+                    worksheet.Range["A" + row].Value = purchase["created"];
+                    worksheet.Range["B" + row].Value = purchase["id"];
+                    worksheet.Range["C" + row].Value = purchase["merchantCategoryCode"];
+                    worksheet.Range["D" + row].Value = purchase["merchantName"];
+                    worksheet.Range["E" + row].Value = purchase["description"];
+                    worksheet.Range["F" + row].Value = purchase["status"];
+                    worksheet.Range["G" + row].Value = purchase["amount"];
+
+                    try
+                    {
+                        foreach (JToken attachment in purchase["attachments"])
+                        {
+                            worksheet.Range["H" + row].Value = attachment["id"] + "," + worksheet.Range["H" + row].Value;
+                        }
+                        worksheet.Range["H" + row].Value = worksheet.Range["H" + row].Value.Substring(0, worksheet.Range["H" + row].Value.Length - 1);
+                    }
+                    catch { }
```

Comment:
Catch vazio? Se tiver em outro lugar e for normal ok. Mas não tem nenhum throw exception?